jina/serve/runtimes/worker/request_handling.py

```diff
@@ -299,43 +355,63 @@ async def handle(
         :param tracing_context: Optional OpenTelemetry tracing context from the originating request.
         :returns: the processed message
         """
+
         # skip executor if endpoints mismatch
-        if (
-            requests[0].header.exec_endpoint not in self._executor.requests
-            and __default_endpoint__ not in self._executor.requests
-        ):
-            self.logger.debug(
-                f'skip executor: mismatch request, exec_endpoint: {requests[0].header.exec_endpoint}, requests: {self._executor.requests}'
-            )
-            return requests[0]
+        exec_endpoint: str = requests[0].header.exec_endpoint
+        if exec_endpoint not in self._executor.requests:
+            if __default_endpoint__ in self._executor.requests:
+                exec_endpoint = __default_endpoint__
+            else:
+                self.logger.debug(
+                    f'skip executor: mismatch request, exec_endpoint: {exec_endpoint}, requests: {self._executor.requests}'
+                )
+                return requests[0]
 
         self._record_request_size_monitoring(requests)
 
         params = self._parse_params(requests[0].parameters, self._executor.metas.name)
-        docs = WorkerRequestHandler.get_docs_from_request(
-            requests,
-            field='docs',
-        )
 
-        # executor logic
-        docs_matrix, docs_map = WorkerRequestHandler._get_docs_matrix_from_request(
-            requests
-        )
-        return_data = await self._executor.__acall__(
-            req_endpoint=requests[0].header.exec_endpoint,
-            docs=docs,
-            parameters=params,
-            docs_matrix=docs_matrix,
-            docs_map=docs_map,
-            tracing_context=tracing_context,
-        )
+        if exec_endpoint in self._batchqueue_config:
+            assert len(requests) == 1, 'dynamic batching does not support no_reduce'
+
+            param_key = str(params)
```

Comment:
I don't think this is a reliable way to generate a key for parameters, try this for instance:
```python
d = {'a': 1, 'b': 'c'}
print(str(d)) # "{'a': 1, 'b': 'c'}"
d = {'b': 'c', 'a': 1}
print(str(d)) #  "{'b': 'c', 'a': 1}"
```
You can refer to this page for ways to hash a dict: https://stackoverflow.com/questions/5884066/hashing-a-dictionary
Ideally, `hash(frozenset(my_dict.items()))` would be great, if we can't use it, we'll have to use `json.dumps(d, sort_keys=True)`

Reply:
agree, this is not reliable